src/Illuminate/Validation/Factory.php

```diff
@@ -186,7 +188,17 @@ protected function addExtensions(Validator $validator)
      */
     public function extend($rule, $extension, $message = null)
     {
-        $this->extensions[$rule] = $extension;
+        if ($extension instanceof RuleContract) {
+            $this->extensions[$rule] = function ($attribute, $value) use ($extension) {
+                return $extension->passes(...func_get_args());
```

Comment:
Why are you passing `... func_get_args()` instead of `$attribute` and `$value` here?

Reply:
good point. The interface describes the method signature, so we can be specific here on the arguments passed on.
PR updated.